relocation R_X86_64_PC32 against absolute symbol `_binary_python_extension_py_size' in section `.text' is disallowed

## Expected Behavior

Building a PRoot binary.

## Actual Behavior

Python extension error when building.

## Steps to Reproduce the Problem

```
make -C src loader.elf loader-m32.elf build.h
make -C src proot
```

## Specifications

  - Proot/Care version: 8c0ccf7db18b5d5ca2f47e1afba7897fb1bb39c0
  - Kernel version: `Linux 49c896f37a82 5.11.0-7612-generic #13~1617215757~20.10~97a8d1a-Ubuntu SMP Thu Apr 1 21:09:17 UTC 2 x86_64 GNU/Linux`
  - Host distribution: Pop!_OS v20.10
  - Guest distribution: Alpine v3.13.0

## Command Output

```terminal
make -C src proot
make: Entering directory '/usr/src/proot/src'
  LD	proot
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: extension/python/python.o: relocation R_X86_64_PC32 against absolute symbol `_binary_python_extension_py_size' in section `.text' is disallowed
collect2: error: ld returned 1 exit status
make: *** [GNUmakefile:179: proot] Error 1
make: Leaving directory '/usr/src/proot/src'
```

--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -17,7 +17,8 @@ PYTHON   = python3
 
 HAS_SWIG := $(shell swig -version 2>/dev/null)
 PYTHON_MAJOR_VERSION = $(shell ${PYTHON} -c "import sys; print(sys.version_info.major)" 2>/dev/null)
-HAS_PYTHON_CONFIG := $(shell ${PYTHON}-config --ldflags 2>/dev/null)
+PYTHON_EMBED = $(shell ${PYTHON} -c "import sys; print('--embed' if sys.hexversion > 0x03080000 else '')" 2>/dev/null)
+HAS_PYTHON_CONFIG := $(shell ${PYTHON}-config --ldflags ${PYTHON_EMBED} 2>/dev/null)
 
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I. -I$(VPATH)
 CFLAGS   += -g -Wall -Wextra -O2
@@ -205,7 +206,7 @@ cli/%-licenses.o: licenses cli/cli.o
 
 define build_python_extension
 CPPFLAGS 	+= $(shell ${PYTHON}-config --includes)
-LDFLAGS 	+= $(shell ${PYTHON}-config --ldflags)
+LDFLAGS 	+= $(shell ${PYTHON}-config --ldflags ${PYTHON_EMBED})
 SWIG     	= swig
 quiet_SWIG 	= @echo "  SWIG	$$@"; swig
 SWIG_OPT 	= -python

--- a/src/cli/proot.h
+++ b/src/cli/proot.h
@@ -269,7 +269,7 @@ Copyright (C) 2021 PRoot Developers, licensed under GPL v2 or later.",
 	  .arguments = {
 		{ .name = "-P", .separator = ' ', .value = "string" },
 		{ .name = NULL, .separator = '\0', .value = NULL } },
-	  .handler = handle_option_p,
+	  .handler = handle_option_P,
 	  .description = "Allow to access tracee information from python (experimental).",
 	  .detail = "\tThis option allow to launch a python script as an extension (experimental).",
 	},

--- a/src/extension/python/python.c
+++ b/src/extension/python/python.c
@@ -23,14 +23,14 @@ extern void PyInit__proot(void);
 
 /* create python files */
 extern unsigned char _binary_python_extension_py_start;
-extern unsigned char _binary_python_extension_py_size;
+extern unsigned char _binary_python_extension_py_end;
 extern unsigned char _binary_proot_py_start;
-extern unsigned char _binary_proot_py_size;
+extern unsigned char _binary_proot_py_end;
 
-static int create_python_file(const char *tmp_dir, const char *python_file_name, unsigned char *start_file, unsigned char *size_file)
+static int create_python_file(const char *tmp_dir, const char *python_file_name, unsigned char *start_file, unsigned char *end_file)
 {
 	void *start = (void *) start_file;
-	size_t size = (size_t) size_file;
+	size_t size = end_file - start_file;
 	char python_full_file_name[PATH_MAX];
 	int fd;
 	int status;
@@ -53,14 +53,14 @@ static int create_python_extension(const char *tmp_dir)
 {
 	return create_python_file(tmp_dir, "python_extension.py",
 								&_binary_python_extension_py_start,
-								&_binary_python_extension_py_size);
+								&_binary_python_extension_py_end);
 }
 
 static int create_proot(const char *tmp_dir)
 {
 	return create_python_file(tmp_dir, "proot.py",
 								&_binary_proot_py_start,
-								&_binary_proot_py_size);
+								&_binary_proot_py_end);
 }
 
 /* init python once */

